tests/commands/ironhawk/decr_test.go

```diff
@@ -74,75 +40,24 @@ func TestDECRBY(t *testing.T) {
 	client := getLocalConnection()
 	defer client.Close()
 
-	type SetCommand struct {
-		key string
-		val int64
-	}
-
-	type DecrByCommand struct {
-		key         string
-		decrValue   any
-		expectedVal int64
-		expectedErr string
-	}
-
-	type GetCommand struct {
-		key         string
-		expectedVal int64
-	}
-
-	testCases := []struct {
-		name           string
-		setCommands    []SetCommand
-		decrByCommands []DecrByCommand
-		getCommands    []GetCommand
-	}{
+	testCases := []TestCase{
 		{
 			name: "Decrement multiple keys",
-			setCommands: []SetCommand{
-				{"key1", 3},
-				{"key3", math.MinInt64 + 1},
-			},
-			decrByCommands: []DecrByCommand{
-				{"key1", int64(2), 1, utils.EmptyStr},
-				{"key1", int64(1), 0, utils.EmptyStr},
-				{"key4", int64(1), -1, utils.EmptyStr},
-				{"key3", int64(1), math.MinInt64, utils.EmptyStr},
-				{"key3", int64(math.MinInt64), 0, "ERR increment or decrement would overflow"},
-				{"key5", "abc", 0, "ERR value is not an integer or out of range"},
-			},
-			getCommands: []GetCommand{
-				{"key1", 0},
-				{"key4", -1},
-			},
+			commands: []string{"SET k1 3", "SET k3 " + fmt.Sprint(math.MinInt64+1), "DECRBY k1 2", "DECRBY k1 1", "DECRBY k4 1", "DECRBY k3 1", "DECRBY k3 " + fmt.Sprint(math.MinInt64), "DECRBY k5 abc", "GET k1", "GET k4"},
+			expected: []interface{}{"OK", "OK", 1, 0, -1, math.MinInt64, 0, "ERR value is not an integer or out of range", 0, -1},
+			keysUsed: []string{"k1", "k3", "k4", "k5"},
```

Comment:
Comparison should be with native V* types from `wire` package and not extrating the elemental values. Ref: https://github.com/DiceDB/dice/pull/1508